modules/tensor_mechanics/src/constraints/NodalFrictionalConstraint.C

```diff
@@ -0,0 +1,365 @@
+                                           _friction_coefficient * _normal_force))
+  {
+    current_force = _friction_coefficient * _normal_force * current_force / std::abs(current_force);
+    printf("yielded: %u \n", _i);
+  }
+
+  switch (type)
+  {
+    case Moose::Slave:
+      return current_force;
+    case Moose::Master:
+      return -current_force;
+  }
+  return 0;
+}
+
+void
+NodalFrictionalConstraint::computeJacobian(SparseMatrix<Number> & jacobian)
+{
+  // Calculate Jacobian enteries and cache those entries along with the row and column indices
+  std::vector<dof_id_type> slavedof = _var.dofIndicesNeighbor();
+  std::vector<dof_id_type> masterdof = _var.dofIndices();
+
+  DenseMatrix<Number> Kee(masterdof.size(), masterdof.size());
+  DenseMatrix<Number> Ken(masterdof.size(), slavedof.size());
+  DenseMatrix<Number> Kne(slavedof.size(), masterdof.size());
+  DenseMatrix<Number> Knn(slavedof.size(), slavedof.size());
+
+  Kee.zero();
+  Ken.zero();
+  Kne.zero();
+  Knn.zero();
+
+  for (_i = 0; _i < slavedof.size(); ++_i)
+  {
+    _j = _master_conn[_i];
+    switch (_formulation)
+    {
+      case Moose::Penalty:
+        Kee(_j, _j) += computeQpJacobian(Moose::MasterMaster);
+        Ken(_j, _i) += computeQpJacobian(Moose::MasterSlave);
+        Kne(_i, _j) += computeQpJacobian(Moose::SlaveMaster);
+        Knn(_i, _i) += computeQpJacobian(Moose::SlaveSlave);
+        break;
+      case Moose::Kinematic:
+        mooseError("This constraint currently works only with penalty formulation.");
+    }
+  }
+  _assembly.cacheJacobianBlock(Kee, masterdof, masterdof, _var.scalingFactor());
+  _assembly.cacheJacobianBlock(Ken, masterdof, slavedof, _var.scalingFactor());
+  _assembly.cacheJacobianBlock(Kne, slavedof, masterdof, _var.scalingFactor());
+  _assembly.cacheJacobianBlock(Knn, slavedof, slavedof, _var.scalingFactor());
+}
+
+Real
+NodalFrictionalConstraint::computeQpJacobian(Moose::ConstraintJacobianType type)
+{
+  Real jac = _tangential_penalty;
+
+  // set jacobian to zero if spring has yielded
+  Real old_force = (_u_slave_old[_i] - _u_master_old[_j]) * _tangential_penalty;
+  if (MooseUtils::absoluteFuzzyGreaterThan(std::abs(old_force),
+                                           _friction_coefficient * _normal_force))
+    old_force = _friction_coefficient * _normal_force * old_force / std::abs(old_force);
+
+  Real current_force = ((_u_slave[_i] - _u_slave_old[_i]) - (_u_master[_j] - _u_master_old[_j])) *
+                           _tangential_penalty +
+                       old_force;
+  if (MooseUtils::absoluteFuzzyGreaterThan(std::abs(current_force),
+                                           _friction_coefficient * _normal_force))
+    jac = 0.0;
+
+  switch (type)
+  {
+    case Moose::SlaveSlave:
+      return jac;
+    case Moose::SlaveMaster:
+      return -jac;
+    case Moose::MasterMaster:
+      return jac;
+    case Moose::MasterSlave:
+      return -jac;
```

Comment:
Add this to get rid of the compile error:
```
       default:
         mooseError("Invalid type");

Reply:
I'll add this to both places and repush.